rust/datafusion/src/physical_plan/sort.rs

```diff
@@ -156,12 +156,14 @@ fn sort_batches(
     )?;
 
     // sort combined record batch
+    // TODO: pushup the limit expression to sort
```

Comment:
The way I have seen this done in other systems is a rewrite pass a `(Sort) -> (Limit)` combination into a new operator `TopK`